src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestUpdate.java

```diff
@@ -258,12 +257,21 @@ String contentOf(String path) throws IOException {
             return FileUtils.readFileToString(file(path), "UTF-8");
         }
 
+        private CliGitAPIImpl cachedCliGitAPIImpl = null;
```

Comment:
Why are you bothering with a cache of a flyweight object in a unit test?

Reply:
> Why are you bothering with a cache of a flyweight object in a unit test?

There are tests that fail unless the same cgit object is returned for multiple calls to a WorkingArea object.  The case that inspired it was JGit as the tested implementation, but the test needs to use CGit features to create submodules before exercising JGit to confirm the submodule checkout works as expected.  The caching makes the change "sticky".

I can add that information as a comment on the cache so that my future self and others can read the rationale in the source code.

Reply:
Yes a comment would be helpful, since the term “cache” is usually understood as solely a performance optimization.

Reply:
Done in 0a828e3f3caf241efe3b16d51d5038a258035c36